radix-engine/src/system/node_properties.rs

```diff
@@ -103,18 +104,17 @@ impl VisibilityProperties {
 
         // TODO: Cleanup and reduce to least privilege
         match (mode, offset) {
-            (ExecutionMode::Kernel, ..) => false, // Protect ourselves!
+            (ExecutionMode::Kernel, offset) => match offset {
+                SubstateOffset::Global(GlobalOffset::Global) => read_only,
+                _ => false, // Protect ourselves!
+            },
```

Comment:
@talekhinezh I removed `Deref` execution mode so to get rid of the two public `deref` methods in `Kernel`, understanding that we're going to ditch the deref idea anyway.